### PR TITLE
fix: Always read http request payloads

### PR DESCRIPTION
**Pull request by untitaker**
See code comments for rationale.

### CODE REVIEW

File: server/src/middlewares.rs [hunk 1]
```diff
@@ -72,3 +73,20 @@ impl<S> Middleware<S> for ErrorHandlers {
...
+/// Read request before returning response. This is not required for RFC compliance, however:
+///
+/// * a lot of clients are not able to deal with unread request payloads
+/// * in HTTP over unix domain sockets, not reading the request payload might cause the client's
+///   write() to block forever due to a filled up socket buffer
+pub struct ReadRequestMiddleware;
+
+impl<S> Middleware<S> for ReadRequestMiddleware {
+    fn response(&self, req: &HttpRequest<S>, resp: HttpResponse) -> Result<Response, Error> {
+        let consume_request = req.payload().for_each(|_| Ok(()));
```
Comment (mitsuhiko) [inline server/src/middlewares.rs:86]: what happens if someone else already read the payload earlier. Does this silently do nothing?

Comment (untitaker) [reply server/src/middlewares.rs:86]: I didn't look into how actix works here but all tests should fail otherwise.

**Review by mitsuhiko — APPROVED** [review]

**Comment by mitsuhiko** [comment]
I think this is fine but in the past I was trying to only exhaust the input streams on endpoints I actually care about responding well. If someone wants to push a 2GB file to a page that is a known 404 I'm okay with that request failing a horrible death.

**Comment by untitaker** [comment]
Since those bad responses are creating 503s in our infra (which is completely misleading for alerting) I think even for those cases this is the better behavior. Ideally a 2GB payload doesn't make it all the way through our infra at all.

**Comment by jan-auer** [comment]
Late to the party here, as always. Just stopping by to say that this is ❤️

**Pull request merged**